### PR TITLE
Fixes #26661 - don't ship fact-plugin on modern Fedora and EL

### PR DESCRIPTION
**Pull request by evgeni**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.21
* [ ] 1.20

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: packages/client/katello-host-tools/katello-host-tools.spec [hunk 1]
```diff
@@ -8,14 +8,15 @@
 
 %global build_agent (0%{?suse_version} == 0) && ((0%{?fedora} > 28) || (0%{?rhel} > 0))
 %global legacy_agent (0%{?rhel} == 5) || (0%{?rhel} == 6)
+%global build_fact_plugin ((0%{?rhel} > 0 && 0%{?rhel} <= 7) || (0%{?fedora} > 0 && 0%{?fedora} <= 24))
```
Comment (ekohl) [inline packages/client/katello-host-tools/katello-host-tools.spec]: We don't support Fedora <= 24 anymore so should we leave that out?

Comment (evgeni) [reply packages/client/katello-host-tools/katello-host-tools.spec]: This sounds like a good idea, yeah.

File: packages/client/katello-host-tools/katello-host-tools.spec [hunk 2]
```diff
@@ -8,14 +8,15 @@
 
 %global build_agent (0%{?suse_version} == 0) && ((0%{?fedora} > 28) || (0%{?rhel} > 0))
 %global legacy_agent (0%{?rhel} == 5) || (0%{?rhel} == 6)
+%global build_fact_plugin (0%{?rhel} > 0 && 0%{?rhel} <= 7)
```
Comment (ekohl) [inline packages/client/katello-host-tools/katello-host-tools.spec]: After reading this, I do wonder about Suse.

Comment (evgeni) [reply packages/client/katello-host-tools/katello-host-tools.spec]: I'll leave that as an exercise to the reader, errr, someone who knows more about our SUSE bits.

Comment (evgeni) [reply packages/client/katello-host-tools/katello-host-tools.spec]: We ship our own sub-man for suse: http://koji.katello.org/koji/packageinfo?packageID=1006, so we should be good?

@jturel might know more

Comment (jturel) [reply packages/client/katello-host-tools/katello-host-tools.spec]: The subman version we ship for suse doesn't need our workaround - so this is good! :taco:

**Pull request merged**